### PR TITLE
Progress

### PR DESCRIPTION
**Pull request by SimonDegrafKern**
PRs:
- https://github.com/code-kern-ai/refinery-ui/pull/137
- https://github.com/code-kern-ai/refinery-gateway/pull/129
- https://github.com/code-kern-ai/refinery-submodule-model/pull/42
- https://github.com/code-kern-ai/refinery-ac-exec-env/pull/19

### CODE REVIEW

**Comment by JWittmeyer** [comment]
why is this branch name `progress` while the others are `ac-progress-bar`

- [ ] no need to change but I acknowledge that linked repository PRs should use the same branch name


Edit: Yeah, this is the plan and I try to commit to it more in the future. Original there was a branch wit h the same name `ac-progress-bar` but then GitHub went wild with their outages which destroyed my branch to a state where I was not able to recover, so I created a new branch as solution, but should have better communicated this!!!

_________________________________
works for me as explanation

File: run_ac.py [hunk 1]
```diff
@@ -103,7 +105,15 @@ def parse_data_to_record_dict(record_chunk):
 
     print("Running attribute calculation.")
     calculated_attribute_by_record_id = {}
+    idx = 0
+    progress_size = 500
```
Comment (JWittmeyer) [inline run_ac.py]: in lf exec env we use chunks of 100.

File: run_ac.py [hunk 2]
```diff
@@ -55,6 +55,8 @@ def __check_data_type_text(attr_value):
         return False
     return True
 
+def __print_progress(progress: float) -> None:
+    print(f":progress:{progress}", flush=True)
```
Comment (JWittmeyer) [inline run_ac.py]: might be wrong but i think there is a : to much in the beginning.

File: run_ac.py (hunk 1 above)
Comment (SimonDegrafKern) [reply run_ac.py]: Will change to 100

File: run_ac.py (hunk 2 above)
Comment (SimonDegrafKern) [reply run_ac.py]: I changed it to this so I could just split with ":" as a seperator and then get the progress with values[-1] if values[-2] == "progress"

Comment (JWittmeyer) [reply run_ac.py]: not the case anymore the code in gateway reads 
`if ":progress:" in log:`

Comment (SimonDegrafKern) [reply run_ac.py]: Yes I can remove the first ":", I think it does not cause any problems but it does not benefit and therefore can be removed

**Review by JWittmeyer — APPROVED** [review]

**Pull request merged**